signer/src/stacks/contracts.rs

```diff
@@ -987,11 +1070,118 @@ impl AsContractCall for RejectWithdrawalV1 {
+        if self.deployer != req_ctx.deployer {
+            return Err(WithdrawalRejectErrorMsg::DeployerMismatch.into_error(req_ctx, self));
+        }
+
+        // 1. The request exists. Check whether the associated withdrawal request transaction
+        //    is confirmed on the canonical stacks blockchain. Fail the withdrawal request if
+        //    it is not on the canonical stacks blockchain.
+        //
+        // 2. The double spend check. Check whether the qualified request ID is in the
+        //    bitcoin_withdrawals_outputs table, and that any of the associated txids are
+        //    confirmed on the canonical bitcoin blockchain. Fail the withdrawal request if
+        //    such a transaction was found.
+
+        let stacks_chain_tip = ctx
+            .get_storage()
+            .get_stacks_chain_tip(&req_ctx.chain_tip.block_hash)
+            .await
+            .unwrap()
+            .unwrap();
+        let Some(report) = ctx
+            .get_storage()
+            .get_withdrawal_request_report(
+                &req_ctx.chain_tip.block_hash,
+                &stacks_chain_tip.block_hash,
+                &self.id,
+                &ctx.config().signer.public_key(),
+            )
+            .await
+            .map_err(|_| WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self))?
+        else {
+            return Err(WithdrawalRejectErrorMsg::RequestMissing.into_error(req_ctx, self));
+        };
+
+        match report.status {
+            WithdrawalRequestStatus::Confirmed => (),
+            WithdrawalRequestStatus::Fulfilled(_txid) => {
+                // fails #2
+                return Err(
+                    WithdrawalRejectErrorMsg::WithdrawalRequestFulfilled.into_error(req_ctx, self)
+                );
+            }
+            WithdrawalRequestStatus::Unconfirmed => {
+                // fails #1
+                return Err(WithdrawalRejectErrorMsg::WithdrawalRequestUnconfirmed
+                    .into_error(req_ctx, self));
+            }
+        }
+
+        // 3. The request is final. Check that we've seen six new bitcoin blocks since observing
+        //    the bitcoin anchor block associated with the Stacks block confirming the withdrawal
+        //    request transaction. Fail the withdrawal request if we’ve observed less than 6
+        //    bitcoin blocks since the anchor block.
+
+        let request_block_height = report.block_height;
+
+        let Some(tip_block_hash) = ctx.get_storage().get_bitcoin_canonical_chain_tip().await?
+        else {
+            return Err(WithdrawalRejectErrorMsg::NoBitcoinChainTip.into_error(req_ctx, self));
+        };
+
+        let Some(tip_bitcoin_block) = ctx.get_storage().get_bitcoin_block(&tip_block_hash).await?
+        else {
+            return Err(WithdrawalRejectErrorMsg::MissingBitcoinBlock.into_error(req_ctx, self));
+        };
+
+        let tip_block_height = tip_bitcoin_block.block_height;
```

Comment:
These are all unnecessary. You should use the chain tip in the `ReqContext`.

Reply:
Done